ui/packages/app/web/src/pages/testpage.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  ParcaContextProvider,
+  Button,
+  useParcaContext,
+  parcaContextDefaultValue,
+  ParcaContextReducer,
+  sendPerfEvent,
+} from '@parca/components';
+import {useReducer} from 'react';
+
+const ChildComponent = () => {
+  const {dispatch} = useParcaContext();
+
+  const sendSomeEvent = () => {
+    console.log('Sending event');
+    dispatch(sendPerfEvent('Some event data here'));
+  };
+
+  return (
+    <>
+      <p>some text here</p>
+      <Button onClick={() => sendSomeEvent()}>Send some perf event</Button>
+    </>
+  );
+};
+
+const Testpage = () => {
+  const [parcaContextState, dispatchToParcaContext] = useReducer(
+    ParcaContextReducer,
+    parcaContextDefaultValue
+  );
+
+  console.log('parcaContextState', parcaContextState.trackedPerfEvents);
+
+  return (
+    <ParcaContextProvider
+      value={{
+        loader: <p>loading...</p>,
+        dispatch: dispatchToParcaContext,
```

Comment:
Instead of passing two fields, for state and setState for the passing the data, should we simplify it to a simple callback whenever there is a perf data? 

Like:
```
<ParcaContextProvider
      value={{
        loader: <p>loading...</p>,
        perf: {
          onRender: // handler that can handler React Profiler API.
          ... // other perf handler that we need to support
         }
      }}
>
...
</ParcaContextProvider>
```
I feel there is there would be less need to send back the data into the child components? 

Wdyt?